src/env-inl.h

```diff
@@ -25,7 +34,6 @@ inline Environment::IsolateData* Environment::IsolateData::GetOrCreate(
     isolate_data = new IsolateData(isolate, loop);
     isolate->SetData(kIsolateSlot, isolate_data);
   }
-  isolate_data->ref_count_ += 1;
```

Comment:
This makes GetOrCreate() rather dangerous to use.  Can I suggest that you remove Get(), GetOrCreate() and Put() and do all management inside the Scope constructor and destructor, with a method or dereference operator to get the raw `IsolateData*`?